### ISSUE TITLE
Quest error messages spamming console

### ISSUE THREAD

**Issue by sweatingbullets67** [PikaMug/Quests#1689]
**What is your Quests version / build number (do _not_ say "latest")?** Version 4.0.2

**Describe the problem and how to replicate it. Any console errors to post _using a pastebin_?** 
-"[16:48:41 ERROR]: [Quests] currentStage was null for 6e0f5881-9cd5-4feb-841d-804ea4a23d44 on chat for quest Local Crawlers" 
-"[17:15:29 ERROR]: Could not pass event BlockDamageEvent to Quests v4.0.2-b238" 
   these are example messages of what I get any time I make a new quest. I've made 33 quests before this without any issues, now when I make a quest, these error messages spam the console. This also occurs when I duplicate an existing quest, changing basic details like name of the quest and the starting NPC.
  Also if it helps. Despite the error messages, the quests still work fine. It's just flooding the console and we are concerned of just not knowing what the issue is.

**If applicable, can you provide an example quest from quests.yml _using a pastebin_?**
Sorry, I don't know what you mean by using a pastebin, but here's one of the quests that are spamming the console.
```
 custom2:
    name: Crawling Critters
    ask-message: Kill 10 spiders
    finish-message: Well done!
    npc-giver-id: 39
    stages:
      ordered:
        '1':
          mobs-to-kill:
          - Spider
          mob-amounts:
          - 10
          start-message: kill 10 spiders
    rewards:
      quest-points: 1
    planner:
      cooldown: 900
```

**Comment by PikaMug** [comment]
Thanks for the report. What do you mean by “duplicate an existing quest”? This is not a function of the plugin.

Also, do you experience any problems associated with the error? Is anything not working correctly?

**Comment by sweatingbullets67** [comment]
So the quest example I provided where a person has to kill 10 spiders, that is formatted the exact same way as a working quest, as in same actions, rewards, etc... only difference is the name and NPC that it starts with. 

To my knowledge, no problems have occurred thus far outside of the console being blown up. We have these error messages under the console spam block for now, but I don't want to risk making more quests if there's an issue I'm just over looking

**Comment by PikaMug** [comment]
I understand you have two near-identical quests, but what method did you use to "duplicate" it? If you simply copy/paste'd within the _quests.yml_ file (not supported), there could be anything wrong from invalid formatting to duplicate quest IDs. It would be necessary to see the file in its entirety to narrow the cause of the issue.

**Comment by sweatingbullets67** [comment]
I used /quests editor to make all the quests. Also these identical quests are just apart of it, I made a quest to kill a wandering trader (only one), got similar error messages. There's a list, and I can actively make it happen again by simply making a quest.

**Comment by PikaMug** [comment]
It's still not clear to me what you mean by "just a apart of it," are you saying you created a quest in the editor, saved/reloaded, and then created a functionally identical quest, again within the editor? If so I'm puzzled by your use of the word "duplicate."

Anyway, a recording won't help in this instance but I appreciate the effort. I would need to see the _quests.yml_ file in its entirety. If you're concerned about sharing it, most pastebin sites let you set an auto-delete timer. A copy of your server's startup log would be helpful, too, since part of your error...

`[17:15:29 ERROR]: Could not pass event BlockDamageEvent to Quests v4.0.2-b238`

...has nothing to do with killing mobs, as in the quest you shared.

**Comment by sweatingbullets67** [comment]
And by "apart of it" I was trying to provide as much possibly needed info as I could. All new  quests that are created will create those 2 error messages. I figured informing that identical quests with the same actions, conditions, etc... may help in narrowing what ever the problem is. 

As of the startup log, I need to wait for owner, I'm just admin. As I said earlier I'm not really tech smart, but he asked me to take care of the quests so I need him to give me, or tell me where to get that file.

**Comment by sweatingbullets67** [comment]
I have never edited the quests.yml and with duplication I mean that I have created a different, but identical new quest in the in-game editor
Without deleting the old one.

**Comment by sweatingbullets67** [comment]
[Questyml.txt](https://github.com/PikaMug/Quests/files/6455629/Questyml.txt)

**Comment by PikaMug** [comment]
Thanks for clarifying, and for providing those files. I've loaded the quests file on my offline test server, but was unable to replicate the issue. Furthermore, the file you provided does not contain either the "Local Crawlers" quest nor the "Crawling Critters" quest from your initial post. In fact, the quest with an ID of `custom2` is an entirely different quest ("Bill's lunch"). I tried swapping the two, but was still unable to produce an error.

Given the startup file looks fine and you say there's no negative effects to gameplay, I'll just remove the error output. However, bear in mind that this error shouldn't have appeared in the first place, so if problems do eventually arise, there may be little to nothing I can do given the lack of debug information.
